### PR TITLE
docs: add RFC for MenuItem refactoring

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-3163]

### Description

RFC for refactoring of MenuItem

### How to test

- discuss the document in review

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-3163]: https://toptal-core.atlassian.net/browse/FX-3163?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by TomasSlama** [comment]
@toptal-anvil ping reviewers

File: docs/decisions/10-menu-item.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+## Proposal
+
+Instead of creating abstraction of Menu.Item, we would create 3 different components.
+1. `Menu.Item` - current solution used in Dropdown, Select, Autocomplete, Page.TopBarMenu. They share the same design.
+2. `Menu.SidebarItem` - instead of overriding lot of styles of Menu.Item, we would create new component
+3. `Menu.AccountItem` - instead of overriding lot of styles of Menu.Item, we would create new component
+
+### Benefits
+
+- Instead of one complex component, we would have three very simple.
```
Comment (ozgurkececioglu) [inline docs/decisions/10-menu-item.md:16]:
```suggestion
- Instead of one complex component, we would have three very simple ones.
```

**Review by ozgurkececioglu — APPROVED** [review]
I'm fine with the proposal. Thanks for this research ❤️

File: docs/decisions/10-menu-item.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+- Updating styles of base Menu.Item would not affect other components with different design.
+- Should be possible to be done without any breaking change
```
Comment (OndrejTuma) [inline docs/decisions/10-menu-item.md:18]: nit:
```suggestion
- Should be possible to do without any breaking changes
```

**Review by OndrejTuma — APPROVED** [review]
It makes sense to me, `SidebarItem` and `AccountItem` are different enough to ask for its own implementation 👍

**Review by ozgurkececioglu — APPROVED** [review]
I'm still thinking that this separation should be handled one day to reduce the complexity of MenuItem 👍

File: docs/decisions/10-menu-item.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+Changes in MenuItem can unexpectedly update multiple components. Currently, MenuItem handles too many things. We can see usage in this [diagram](https://miro.com/app/board/uXjVPOOdjH8=/?share_link_id=687084096239).
+
+## Proposal
+
+Instead of creating abstraction of Menu.Item, we would create multiple different components.
+1. `Menu.DropdownItem` - current solution used in Dropdown, Page.TopBarMenu. They share the same design.
+2. `Menu.SelectItem` - different design and API
+2. `Menu.SidebarItem` - instead of overriding lot of styles of Menu.Item, we would create new component
+3. `Menu.AccountItem` - instead of overriding lot of styles of Menu.Item, we would create new component
```
Comment (dmaklygin) [inline docs/decisions/10-menu-item.md:13]: `AccountItem` for me, seems an extension of `UserBadge + MenuItem` Component. 
Do you know, whether we can reuse it?

Comment (TomasSlama) [reply docs/decisions/10-menu-item.md:13]: @dmaklygin yeah, we already have it in [AccountSelect](https://github.com/toptal/picasso/blob/master/packages/picasso/src/AccountSelect/AccountSelect.tsx). I have looked into implementation details and we will probably don't have to create a new component for that, there is not that many overwrites 🤔

Comment (dmaklygin) [reply docs/decisions/10-menu-item.md]: I can see, that AccountSelect is used only once: https://github.com/toptal/platform/blob/c5b4e5a0749ff44a05b724ababad3e1a5edf0506/app/assets/features/platform/role_selection/components/RoleSelectionScreen/index.jsx#L33

Whether it's intentional to keep it inside Picasso?

Comment (TomasSlama) [reply docs/decisions/10-menu-item.md]: This is actually good point, but do you think it is acceptable to move component from Picasso to platform? 🤔

File: docs/decisions/10-menu-item.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+3. `Menu.SidebarItem` - instead of overriding lot of styles of Menu.Item, we would create new component
+
+### Benefits
+
+- Instead of one complex component that does everything, we would have multiple very simple ones.
+- Updating styles of base Menu.Item would not affect other components with different design.
+
+### Drawbacks and limitations
+
+- Will cause breaking change that would be hard to fix with codemods
```
Comment (OndrejTuma) [inline docs/decisions/10-menu-item.md:21]: I'm thinking, how will this cause breaking change? 🤔 
MenuItem styles should remain the same, no?

Comment (TomasSlama) [reply docs/decisions/10-menu-item.md:21]: @OndrejTuma in my head, we might remove some props from Menu.Item, for example `checkmarked`. Instead Menu.Item will have prop `icon`. Then we would have Select.Item (ignore naming), that would use `<Menu.Item icon={<IconMark16/>} 
 ...plusOtherDifferences />`

File: docs/decisions/10-menu-item.md (hunk 3 above)
Comment (TomasSlama) [reply docs/decisions/10-menu-item.md]: I agree 👍, I will add it to RFC

**Review by dmaklygin — APPROVED** [review]

**Pull request merged**